include/boost/url/impl/url.ipp

```diff
@@ -2351,6 +2355,21 @@ ensure_space(
     if(nparam > 0)
         new_cap += 2 * sizeof(pos_t) *
             nparam;
+
+    static constexpr auto AS =
+        alignof(detail::segments_table_entry);
+    static constexpr auto SS =
+        sizeof(detail::segments_table_entry);
+    new_cap = AS * ((new_cap + AS - 1) / AS) +
+           (nseg * SS);
+
+    static constexpr auto AP =
+        alignof(detail::params_table_entry);
+    static constexpr auto SP =
+        sizeof(detail::params_table_entry);
+    new_cap = AP * ((new_cap + AP - 1) / AP) +
+           (nparam * SP);
```

Comment:
The problem here is we have two tables back to back. That means that any changes in size of the second table are going to cause memmove of the entire first table. I was thinking instead that we have just one table with enough space for the larger of the two sizes, where each element stores a param and a segment offset.

Reply:
> I was thinking instead that we have just one table with enough space for the larger of the two sizes, where each element stores a param and a segment offset.

Mmm... This is possible. In that case, we would need a new name for the `<component>_table_entry`, `<component>_table`, and `const_<component>_table`. 

It's also a trade-off but I'm not sure of all the implications:

- From a perspective of memory, we are obviously always spending more memory, unless `nseg == nparam`, which should be quite rare.
- From a perspective of computational cost, I have the impression the merged table is absorbing the cost of manually moving elements into any operation changing any of the tables, which is more costly than memmoving only one of the tables.
- From the perspective of code complexity, I'm not sure yet, but I have the impression we're going to have lots of ifs and elses to do what memmove would.

Reply:
Well I will leave the decision up to you